### DISCUSSION TITLE
Using gpt-engineer with Azure OpenAI service endpoint

### DISCUSSION THREAD

**Discussion by HermanCodes** [AntonOsika/gpt-engineer#503] [Q&A]
I have an Azure OpenAI endpoint with a GPT-35-Turbo model deployed. Is it possible for me to use this endpoint and key with gpt-engineer?

**Comment by smitzlroy** [comment]
Any insight on this question would be great.

**Comment by machmuel** [comment]
> I have an Azure OpenAI endpoint with a GPT-35-Turbo model deployed. Is it possible for me to use this endpoint and key with gpt-engineer?

I´m also using my own _Azure OpenAI Service_ (https://..... .openai.azure.com/). 
My goal is to use it for training of own data in my own separated area. The used openai library seems actually to also support usage of _Azure OpenAI Service_. 

So, it would be very beneficial (for me and others) to also add support of _Azure OpenAI Service_ as openai.api_type.

I already checked the code to find the code location, where it´d needed to be extended. 
If someone could give me a hint, I´d provide even a prepared pull request for it.

Many thanks in advance!



@smitzlroy

**Comment by magnusahlden** [comment]
> I have an Azure OpenAI endpoint with a GPT-35-Turbo model deployed. Is it possible for me to use this endpoint and key with gpt-engineer?

yes, just use --azure your-azure-url yourdirectory-with-prompt NAME_OF_YOUR_DEPLOYED_MODEL

for me it works with gpt-engineer --azure mga.openai.azure.com ./example/ magnus_model

where magnus_model is the deployed GPT3.5-turbo model
